core/geometry/src/curve/CurveFactory.ts

```diff
@@ -426,101 +432,154 @@ export class CurveFactory {
-   * * In the "open path" case (i.e when `wrapIfPhysicallyClosed` is false or the path does not have matched first and
-   *   last points), the first/last output plane will be at the start/end of the first/last edge and on a perpendicular
-   *   plane.
-   * * In the "closed path" case, if start/edn tangents are not provided in the `options`, then the output plane for the
-   *   first and last point is the bisector of the start and end planes from the "open path" case, and the first/last
-   *   section geometry may be different from `initialSection`.
-   * * The centerline path does NOT have to be planar, however twisting effects effects will appear in the various bisector
-   *   planes.
-   * @param centerline sweep path, e.g., as stroked from a smooth centerline curve.
+   * Sweep the `initialSection` along each segment of the (stroked) `centerline` until it hits the bisector plane at
+   * the next vertex.
+   * * For best results, the caller should place `initialSection` in a plane perpendicular to the `centerline`
+   * start tangent.
+   *   * This plane is commonly (but not necessarily) through the centerline start point itself.
+   *   * To compute the sections, `initialSection` is projected along the start tangent onto the first bisector plane
+   * at the centerline start. The result of this projection will be likewise projected onto the second plane, and so
+   * on in sequence.
+   * * By default, the first/last bisector plane normals are set to the centerline start/end tangents. The caller can
+   * override these with tangents supplied in `options`. If the centerline is physically closed and
+   * `options.wrapIfPhysicallyClosed` is true, the first and last plane normals are averaged and equated.
+   * * The centerline path does NOT have to be planar, however non-planarity will result in twisting of the sections
+   * in the bisector planes.
+   * @param centerline sweep path. Will be stroked if smooth.
    * @param initialSection profile curve to be swept. As noted above, this should be on a plane perpendicular to the
-   * first segment of the centerline.
+   * centerline start tangent.
    * @param options options for computation and output.
-   * @return array of sections, starting with `initialSection` projected along the first edge to the first plane.
+   * @return array of sections, formed from projecting `initialSection` successively onto the bisector planes.
    */
   public static createMiteredSweepSections(
     centerline: IndexedXYZCollection | Point3d[] | CurvePrimitive | CurveChain,
     initialSection: AnyCurve,
     options: MiteredSweepOptions,
   ): SectionSequenceWithPlanes | undefined {
-    if ((centerline instanceof IndexedXYZCollection || Array.isArray(centerline)) && centerline.length < 2)
+    const rail = this.strokeSmoothCurve(centerline, options.strokeOptions);
+    if (!rail)
+      return undefined;
+
+    const planes = PolylineOps.createBisectorPlanesForDistinctPoints(rail.strokes);
```

Comment:
I moved the averaging of first and last planes for a closed rail to `alignFirstAndLastBisectorPlanes`, where we have the tangent overrides available. Tangent overrides come from the `MiteredSweepOptions` if set, otherwise they are computed from a smooth rail, otherwise (if the rail is a polyline) the tangents are read off the planes themselves.